api/game/clueless.py

```diff
@@ -24,56 +29,226 @@ def __init__(self, *args, **kwargs):
         ]
         self.secret_passages = ['study_kitchen', 'lounge_conservatory']
 
+        # TODO connection manager should contain each client's character choice
+          # self.players should be filled & board should be initialized 
+          # after all clients have connected and game starts. 
+        
         # players dict holds client_id: character assignments
         self.players = {}
         self.state = self.initialize_board(self.players)
+    
+
+    def jsonify(self):
+        return json.dumps(self.state)
+
+
+    def get_game_state(self):
+        return self.state
+
 
-    def initialize_board(self, players):
-        """ Places characters in starting locations, generates sceanario, 
+    def initialize_board(self, players: Dict) -> Dict:
+        """ Places characters in starting locations, generates scenario, 
         distributes cards """
 
         # data structs are just placeholders; can/should be changed
         state = {
             'suspect_locations': {},  # dict of suspect: room/hallway loc
             'concealed_scenario': {},
-            'player_cards': {},  # dict of player: player's card list 
+            'player_cards': {},  # dict of client id: player's card list 
             'visible_cards': [],  # list of cards shown to all players
             'turn_order': [],  # player turn order
-            'current_turn': 0,  # whose turn is it
+            'current_turn': (),  # whose turn is it? (client id, charname)
             'suggestion': {}  # holds current suggestion players must disprove
         }
         # starting locations
         state['suspect_locations'] = {
-            'colonel_mustard': ' lounge_dining',
             'miss_scarlet': 'hall_lounge',
             'professor_plum': 'library_study',
             'mr_green': 'ballroom_conservatory',
             'mrs_white': 'kitchen_ballroom',
             'mrs_peacock': 'conservatory_library',
+            'colonel_mustard': ' lounge_dining',
         }
+        # randomly generated case file cards
+        state['concealed_scenario'] = self.create_scenario()
+
+        # shuffle and distribute cards to players
+        state['player_cards'], state['visible_cards'] = self.distribute_cards()
+
+        # set turn order for the game
+        state['turn_order'] = self.generate_turn_order()
+
 
+        # get client id of first player to move
+        for client_id, token in self.players.items():

```

Comment:
remove references to clientid. game logic should be independent of connection logic

Reply:
done